src/fsverity/ioctl.rs

```diff
@@ -63,16 +82,48 @@ pub fn fs_ioc_measure_verity<F: AsFd, H: FsVerityHashValue>(fd: F) -> Result<H>
         digest: H::EMPTY,
     };
 
-    unsafe {
+    let r = unsafe {
         ioctl::ioctl(
             fd,
             ioctl::Updater::<FsIocMeasureVerity, FsVerityDigest<H>>::new(&mut digest),
-        )?;
+        )
+    };
+    match r {
+        Ok(()) => {
+            if digest.digest_algorithm != digest_algorithm {
+                return Err(MeasureVerityError::InvalidDigestAlgorithm {
+                    expected: digest.digest_algorithm,
+                    found: digest_algorithm,
+                });
+            }
+            if digest.digest_size != digest_size {
+                return Err(MeasureVerityError::InvalidDigestSize {
+                    expected: digest.digest_size,
+                });
+            }
+            Ok(Some(digest.digest))
+        }
+        // This function returns Ok(None) if there's no verity digest found.
+        Err(Errno::NODATA | Errno::NOTTY) => Ok(None),
```

Comment:
The fsverity docs say `EOPNOTSUPP` for no kernel support.  I guess I'd count that as "file doesn't have fs-verity".